### ISSUE TITLE
Server: clickable links to rule documentation are invalid for code block tools

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#505]
### rumdl version

0.1.43

### Installation method

mise

### Bug description

When `rumdl server` encounters a violation of a rule, it provides to the LSP client a clickable link for more information about the rule. The link takes the form `https://rumdl.dev/<rule_name>`.

However, when using code block tools, instead of a rule name, the tool name is used instead. This means that `rumdl server` can provide the LSP client a link like `https://rumdl.dev/jq/` (when a JSON problem is found), which is invalid.

### Steps to reproduce

1. Create a Markdown file with the following content (the JSON is invalid):
   ````markdown
   ---
   title: Heading
   ---
   
   Some introductory text.
   
   ```json
   {
     "test":: "test"
   }
   ```
   ````
1. Create a `.rumdl.toml` file with this content:
   ```toml
   [code-block-tools]
   enabled = true
   
   [code-block-tools.languages]
   json = { lint = ["jq"] }
   ```
1. Use `rumdl` in server mode to inspect the file.

### Expected behavior

Only valid URLs should be returned to the LSP client. No URL is better than an invalid URL.

### Actual behavior

The URL `https://rumdl.dev/jq/` is returned to the LSP client.

### Configuration

```toml
[code-block-tools]
enabled = true

[code-block-tools.languages]
json = { lint = ["jq"] }
```

### Operating system

macOS

**Comment by eread** [comment]
@rvben Incidentally, `jq` is listed at https://rumdl.dev/code-block-tools/#built-in-tools as a formatter and not a linter, but it works fine as a linter too when the parse errors are returned to the LSP client.

**Comment by rvben** [comment]
Fixed in v0.1.44. The LSP server now skips documentation URLs for code-block-tool warnings instead of generating invalid links. Thanks for reporting @eread!
